cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/ldap/LdapUserPersistence.java

```diff
@@ -155,37 +155,45 @@ public User addUser(User user)
     public User addUserRequest(UserRequest userRequest, final Principal ownerHttpPrincipal)
         throws UserNotFoundException, TransientException, AccessControlException, UserAlreadyExistsException
     {
-        LdapUserDAO userDAO = null;
         LdapConnections conns = new LdapConnections(this);
+        LdapUserDAO userDAO = null;
+        LdapGroupDAO groupDAO = null;
         User user = null;
+        Group group = null;
         try
         {
             // create the group to be associated with this userRequest
             userDAO = new LdapUserDAO(conns);
-            LdapGroupDAO groupDAO = new LdapGroupDAO(conns, userDAO);
+            groupDAO = new LdapGroupDAO(conns, userDAO);
             LocalAuthority localAuthority = new LocalAuthority();
             URI gmsServiceURI = localAuthority.getServiceURI(Standards.GMS_GROUPS_01.toString());
             GroupURI groupID = new GroupURI(gmsServiceURI, userRequest.getUser().getHttpPrincipal().getName());
-            Group group = new Group(groupID);
+            group = new Group(groupID);
             User groupOwner = userDAO.getAugmentedUser(ownerHttpPrincipal,  false);
             ObjectUtil.setField(group, groupOwner, "owner");
-
-            // ensure that there is no existing group with the same user name
-            boolean groupExists = checkIfGroupExists(group, groupDAO);
-            if (groupExists) {
-                throw new GroupAlreadyExistsException("Group " + group.getID().getName() + " already exists ");
-            }
             
-            // group does not exist, add the userRequest
+            // add the userRequest
             user = userDAO.addUserRequest(userRequest);
+           
+            // add the user to the group
             group.getUserMembers().add(user);
             
             // add the group associated with the userRequest
             groupDAO.addUserAssociatedGroup(group, user.posixDetails.getGid());
         } catch (GroupAlreadyExistsException ex) {
-            // clean up
-            if (user != null) {
-                deleteUserRequest(user.getHttpPrincipal());
+            // generate another userRequest
+        	group.getUserMembers().clear();
+            user = userDAO.addUserRequest(userRequest);
+            group.getUserMembers().add(user);
```

Comment:
Why is there a second attempt to create a user request?

Reply:
When a UserRequest is created, its uid is generated. The same uid is used as the gid. If we detect that the group already exists right after we have just created a UserRequest, it indicates that there is a race condition. If we create another UserRequest (i.e. use a newly generated uid), it's highly unlikely that the group already exists (i.e. a group with the same value as the newly generated uid has just been created). That's the reason for the 2nd attempt to create a user request. I suppose I can synchronize the codes. Just thought that this is simpler and should work most (99.99% ;-) of the time.

Reply:
I incermented build.gradle to 1.1.24 in branch s2738 on July 10. Pat incremened the same build.gradle to 1.1.24 on July 17 and released it to the master branch. This explains why my build.gradle file change did not show up in the PR. I have incremented it again to version 1.1.25.

Reply:
If there is a collision with a newly created uid/gid, then just fail.  Don't try to do it again.

Reply:
The fail should be internal server error (500) so a IllegalStateException

Reply:
Making sure you saw these comments...

Reply:
Since Pat has not released his changes, there is no need for me to increment the build version, so I decremented it to keep it at 1.1.24.

Okay. Will update the code to throw an illegalStateException.

Reply:
Done. Please note that there are changes in service.git/ac that need to be reviewed. Basically I added an integration test there.